### PR TITLE
Allow an option for OTA Requestor to be commissioned externally

### PR DESCRIPTION
**Pull request by selissia**
#### Problem
Need an  an option for the OTA Requestor to be commissioned externally.

#### Change overview
- Add Commissioning  and OperationalCredentials clusters to the Requestor app. 
- Add the udpPort command line parameter to the Requestor app. The logic for the Requestor to proceed with self-commissioning is based on the presence of the udpPort parameter (there's no need for setting the UDP port unless we are skipping self-commissioning) -- this logic is temporary as self-commissioning will be removed from this app in the next few weeks. 
- Add the discriminator command line parameter to the Requestor app.
- Also add Commissioning cluster to the Provider app -- this allows it to be commissioned through the Python Controller (in addition to chip-tool).

#### Testing
Commissioned the Requestor through chip-tool

### CODE REVIEW

File: examples/ota-requestor-app/linux/README.md [hunk 1]
```diff
@@ -33,6 +33,11 @@ In a second terminal:
 -   Code for running a full BDX download exists in BDX
 -   Sends QueryImage command
 -   Takes a peer Node ID as an argument
+-   Takes an optional udpPort command argument which when present disables
```
Comment (carol-apple) [inline examples/ota-requestor-app/linux/README.md:36]: Should update instructions above near line 26. Need to add the two different ways to commission (one for the existing way of calling `chip-tool` once for ota-provider app only and the second way is to call `chip-tool` twice, once for ota-provider, and a second time for ota-requestor)

File: examples/ota-requestor-app/linux/main.cpp [hunk 2]
```diff
@@ -174,6 +178,15 @@ bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier,
             PrintArgError("%s: unable to parse Node ID: %s\n", aProgram, aValue);
         }
         break;
+    case kOptionUdpPort:
+         requestorSecurePort = static_cast<uint16_t>(strtol(aValue, NULL, 0));
+
+         if(requestorSecurePort == 0)
```
Comment (carol-apple) [inline examples/ota-requestor-app/linux/main.cpp:184]: nit: Missing space between `if` and `(`

File: examples/ota-requestor-app/linux/main.cpp [hunk 3]
```diff
@@ -231,25 +248,43 @@ int main(int argc, char * argv[])
...
-    // Until #9518 is fixed, the only way to open a CASE session to another node is to commission it first using the
-    // DeviceController API. Thus, the ota-requestor-app must do self commissioning and then read CASE credentials from persistent
-    // storage to connect to the Provider node. See README.md for instructions.
-    // NOTE: Controller is initialized in this call
-    err = DoExampleSelfCommissioning(mController, &mOpCredsIssuer, &mStorage, mStorage.GetLocalNodeId(), mStorage.GetListenPort());
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(SoftwareUpdate, "example self-commissioning failed: %s", chip::ErrorStr(err)));
+    // When the udpPort command line parameter is not present the Requestor self-commissions and automatically requests
+    // an image from the Provider. When the parameter is present the Requestor initializes like any other application and
+    // does not perform any automatic actions.
+    if(requestorSecurePort != 0)
```
Comment (carol-apple) [inline examples/ota-requestor-app/linux/main.cpp:254]: nit: Missing space between if and (

File: examples/ota-requestor-app/ota-requestor-common/BUILD.gn [hunk 4]
```diff
@@ -35,5 +35,7 @@ chip_data_model("ota-requestor-common") {
     "BDXDownloader.h",
   ]
 
-  public_configs = [ ":config" ]
+  is_server = true
+
+public_configs = [ ":config" ]
```
Comment (carol-apple) [inline examples/ota-requestor-app/ota-requestor-common/BUILD.gn:40]: nit: Missing indentation

File: examples/ota-requestor-app/linux/main.cpp (hunk 3 above)
Comment (selissia) [reply examples/ota-requestor-app/linux/main.cpp]: Fixed

File: examples/ota-requestor-app/linux/main.cpp [hunk 5]
```diff
@@ -231,25 +248,43 @@ int main(int argc, char * argv[])
...
+        {
```
[the remark this answers is not in the document]
Comment (selissia) [reply examples/ota-requestor-app/linux/main.cpp]: Applied auto restyling

File: examples/ota-requestor-app/ota-requestor-common/BUILD.gn (hunk 4 above)
Comment (selissia) [reply examples/ota-requestor-app/ota-requestor-common/BUILD.gn]: Fixed

**Review by holbrookt — APPROVED** [review]

**Review by carol-apple — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**